src/components/shared/helpers/normalizeProps.js

```diff
@@ -0,0 +1,35 @@
+const PRIMITIVE_TYPES = new Set(['string', 'number', 'boolean'])
+
+const createDefaultChangeHandler = onChange => payload => {
+    const value =
+        // ui-core event signature
+        (payload && payload.value) ||
+        // synthetic event
+        (payload && payload.target && payload.target.value) ||
+        // directly usable value
+        (PRIMITIVE_TYPES.has(typeof payload) && payload) ||
+        // ¯\_(ツ)_/¯
+        ''
```

Comment:
Should we throw an error here? Since this might potentially be hard to debug for a user if we're supplying an empty string when all else fails.

Reply:
Very good point!